### PR TITLE
Replace SecretService and ConfigService with @t3-oss/env-core

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
Now env variables are validated via zod.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [x] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [x] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings and errors
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

File: services/workflows-service/src/env.ts [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+import 'dotenv/config';
+import { createEnv } from '@t3-oss/env-core';
+import { z } from 'zod';
 
-export const JWT_EXPIRATION = 'JWT_EXPIRATION';
+export const env = createEnv({
+  /*
+   * clientPrefix is required.
+   */
+  clientPrefix: 'PUBLIC_',
```
Comment (alonp99) [inline services/workflows-service/src/env.ts:9]: What `clientPrefix` means in that context?

File: services/workflows-service/src/env.ts [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+  server: {
+    NODE_ENV: z.enum(['development', 'production', 'test']),
+    ENV_FILE_NAME: z.string().optional(),
+    BCRYPT_SALT: z.coerce.number().int().nonnegative(),
```
Comment (alonp99) [inline services/workflows-service/src/env.ts:13]: This validation dosent match the original, string is acceptable

**Review by alonp99 — CHANGES_REQUESTED** [review]
Good job! See SALT validation comment

File: services/workflows-service/src/env.ts (hunk 1 above)
Comment (Omri-Levy) [reply services/workflows-service/src/env.ts:9]: It's for frameworks like NextJS where only env variables prefixed with PUBLIC_ are available client-side.

File: services/workflows-service/src/env.ts (hunk 2 above)
Comment (Omri-Levy) [reply services/workflows-service/src/env.ts:13]: The salt tests are checking that it is a number and is above 0. There was also parseInt in use prior to my changes (zod now converts to number)

**Review by alonp99 — APPROVED** [review]

**Review by alonp99 — APPROVED** [review]

**Review by alonp99 — APPROVED** [review]

**Pull request merged**